clippy_lints/src/excessive_precision.rs

```diff
@@ -6,31 +6,33 @@ use rustc_errors::Applicability;
 use rustc_hir as hir;
 use rustc_lint::{LateContext, LateLintPass};
 use rustc_session::{declare_lint_pass, declare_tool_lint};
-use std::{f32, f64, fmt};
+use rustc_span::symbol::Symbol;
+use std::f32;
+use std::f64;
+use std::fmt;
 use syntax::ast::*;
 
 declare_clippy_lint! {
     /// **What it does:** Checks for float literals with a precision greater
-    /// than that supported by the underlying type.
+    /// than that supported by the underlying type
     ///
-    /// **Why is this bad?** Rust will silently lose precision during conversion
-    /// to a float.
+    /// **Why is this bad?** Rust will truncate the literal silently.
     ///
     /// **Known problems:** None.
     ///
     /// **Example:**
     ///
     /// ```rust
     /// // Bad
-    /// let a: f32 = 0.123_456_789_9; // 0.123_456_789
-    /// let b: f32 = 16_777_217.0; // 16_777_216.0
+    /// let v: f32 = 0.123_456_789_9;
+    /// println!("{}", v); //  0.123_456_789
     ///
     /// // Good
-    /// let a: f64 = 0.123_456_789_9;
-    /// let b: f64 = 16_777_216.0;
+    /// let v: f64 = 0.123_456_789_9;
+    /// println!("{}", v); //  0.123_456_789_9
     /// ```
     pub EXCESSIVE_PRECISION,
-    correctness,
+    style,
```

Comment:
Yeah, let's leave it in `style`. deny-by-default is probably to harsh for this lint.

Reply:
Ok sounds good. `style` just doesn't feel like the right place to put it but we can revisit this later.